### PR TITLE
[TA-7699][TA-7474] Make addressLookup compatible with donation form and handle ajax errors

### PR DESCRIPTION
**Pull request by benoneal**

### CODE REVIEW

File: forms/AddressFieldset/__tests__/AddressFieldset-test.js [hunk 1]
```diff
@@ -1,35 +1,36 @@
...
+  paf_validated: true,
+  country_name: 'Australia',
+  locality: '',
+  region: '',
+  postal_code: ''
+}
 
 describe('AddressFieldset', () => {
-  describe('#handleChange()', () => {
+  describe('#onFieldChange()', () => {
```
Comment (bradparker) [inline forms/AddressFieldset/__tests__/AddressFieldset-test.js]: Much better!

File: forms/AddressFieldsetWithLookup/index.js [hunk 2]
```diff
@@ -2,96 +2,116 @@
...
-    let countryCode = this.refs.lookup.state.selectedCountry.value
-    this.setState({
-      countryCode,
-      address
-    }, () => {
-      this.refs.fieldset.validate()
-      this.props.onChange(address)
-    })
+  handleError(err) {
+    if (typeof err !== 'boolean') {
```
Comment (bradparker) [inline forms/AddressFieldsetWithLookup/index.js]: Is this to say that network errors are objects / strings?

Comment (benoneal) [inline forms/AddressFieldsetWithLookup/index.js]: yup.

Comment (bradparker) [inline forms/AddressFieldsetWithLookup/index.js]: Maybe we should think about casting values before they make it into functions?
Just as a pattern for the future, pretend like this stuff is type checked... ?

**Pull request merged**